components/tournaments/Preview/MyTournamentPreviews.tsx

```diff
@@ -31,46 +32,103 @@ export function MyTournamentPreviews (props: MyTournamentPreviewsProps) {
           <CardDescription>Click New Tournament to get started!</CardDescription>
         </CardHeader>
       </Card>
-    )
+    );
   }
 
-  const availableTournamentCategories: TournamentCategoryTab[] = allTournamentCategoryTabs.filter((cat) => (cat === 'all') || tournaments?.some((tournament) => tournament.category === cat));
+  const availableTournamentCategories: TournamentCategoryTab[] = allTournamentCategoryTabs.filter(
```

Comment:
can you put these in useMemo/useCallback hooks before the return? else they will rerun on every render/child render